config/config.go

```diff
@@ -241,16 +241,26 @@ func (t *GDPRTimeouts) ActiveTimeout() time.Duration {
 // TCF2 defines the TCF2 specific configurations for GDPR
 type TCF2 struct {
 	Enabled             bool                `mapstructure:"enabled"`
-	Purpose1            PurposeDetail       `mapstructure:"purpose1"`
-	Purpose2            PurposeDetail       `mapstructure:"purpose2"`
-	Purpose7            PurposeDetail       `mapstructure:"purpose7"`
-	SpecialPurpose1     PurposeDetail       `mapstructure:"special_purpose1"`
+	Purpose1            Purpose             `mapstructure:"purpose1"`
+	Purpose2            Purpose             `mapstructure:"purpose2"`
+	Purpose3            Purpose             `mapstructure:"purpose3"`
+	Purpose4            Purpose             `mapstructure:"purpose4"`
+	Purpose5            Purpose             `mapstructure:"purpose5"`
+	Purpose6            Purpose             `mapstructure:"purpose6"`
+	Purpose7            Purpose             `mapstructure:"purpose7"`
+	Purpose8            Purpose             `mapstructure:"purpose8"`
+	Purpose9            Purpose             `mapstructure:"purpose9"`
+	Purpose10           Purpose             `mapstructure:"purpose10"`
+	SpecialPurpose1     Purpose             `mapstructure:"special_purpose1"`
 	PurposeOneTreatment PurposeOneTreatment `mapstructure:"purpose_one_treatment"`
 }
 
 // Making a purpose struct so purpose specific details can be added later.
-type PurposeDetail struct {
+type Purpose struct {
 	Enabled bool `mapstructure:"enabled"`
+	// Array of vendor exceptions that is used to create the hash table VendorExceptionMap so vendor names can be instantly accessed
+	VendorExceptions   []openrtb_ext.BidderName `mapstructure:"vendor_exceptions"`
+	VendorExceptionMap map[openrtb_ext.BidderName]bool
```

Comment:
To simulate a hash set, please use `map[openrtb_ext.BidderName]struct{}` instead. The empty struct uses no memory.